app/src/main/java/com/github/onlynotesswent/utils/ReUsableComposables.kt

```diff
@@ -0,0 +1,498 @@
+ *
+ * @param text The text to be displayed in the dropdown menu item.
+ * @param icon The icon to be displayed in the dropdown menu item.
+ * @param onClick The action to be invoked when the dropdown menu item is clicked.
+ * @param modifier The modifier for the dropdown menu item.
+ */
+data class CustomDropDownMenuItem(
+    val text: @Composable () -> Unit,
+    val icon: @Composable () -> Unit,
+    val onClick: () -> Unit,
+    val modifier: Modifier = Modifier
+)
+
+/**
+ * A composable function that displays an `OutlinedTextField` with an optional trailing icon.
+ *
+ * @param value The current value of the text field.
+ * @param onValueChange The callback that is triggered when the value changes.
+ * @param label The label to be displayed inside the text field container.
+ * @param placeholder The placeholder to be displayed when the text field is in focus and the input
+ *   text is empty.
+ * @param modifier The modifier to be applied to the `OutlinedTextField`.
+ * @param trailingIcon An optional trailing icon displayed at the end of the text field container.
+ */
+@Composable
+fun NoteDataTextField(
+    value: String,
+    onValueChange: (String) -> Unit,
+    label: String,
+    placeholder: String,
+    modifier: Modifier = Modifier,
+    trailingIcon: @Composable (() -> Unit)? = null
+) {
+  OutlinedTextField(
+      value = value,
+      onValueChange = onValueChange,
+      label = { Text(label) },
+      placeholder = { Text(placeholder) },
+      modifier = modifier,
+      trailingIcon = trailingIcon,
+      colors =
+          TextFieldDefaults.colors(
+              focusedIndicatorColor = MaterialTheme.colorScheme.primary,
+              unfocusedIndicatorColor = MaterialTheme.colorScheme.onBackground,
+              focusedContainerColor = MaterialTheme.colorScheme.background,
+              unfocusedContainerColor = MaterialTheme.colorScheme.background))
+}
+
+/**
+ * A composable function that displays the top app bar for the screen. It is composed of an icon
+ * button and a title.
+ *
+ * @param title The title to be displayed in the top app bar.
+ * @param titleTestTag The test tag for the title.
+ * @param onBackClick The callback to be invoked when the back button is clicked.
+ * @param icon The icon to be displayed in the top app bar.
+ * @param iconTestTag The test tag for the icon.
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun ScreenTopBar(
+    title: String,
+    titleTestTag: String,
+    onBackClick: () -> Unit,
+    icon: @Composable () -> Unit,
+    iconTestTag: String
+) {
+  TopAppBar(
+      colors =
+          TopAppBarDefaults.topAppBarColors(containerColor = MaterialTheme.colorScheme.surface),
+      title = {
+        Row(
+            modifier = Modifier.fillMaxWidth(),
+            verticalAlignment = Alignment.CenterVertically,
+            horizontalArrangement = Arrangement.Center) {
+              Spacer(modifier = Modifier.weight(1.4f))
```

Comment:
Using a box with a center alignment would maybe be a better idea than using Spacers.